pkg/manager/internal/controllers/organizationroletemplate_controller.go

```diff
@@ -96,3 +156,71 @@ func (r *OrganizationRoleTemplateReconciler) handleDeletion(ctx context.Context,
 
 	return nil
 }
+
+func (r *OrganizationRoleTemplateReconciler) reconcileRBACForOrganization(ctx context.Context, organizationRoleTemplate *corev1alpha1.OrganizationRoleTemplate, organization *corev1alpha1.Organization) error {
+	// Reconcile Role.
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      organizationRoleTemplate.Name,
+			Namespace: organization.Status.Namespace.Name,
+		},
+		Rules: organizationRoleTemplate.Spec.Rules,
+	}
+	if err := r.reconcileRole(ctx, role, organizationRoleTemplate); err != nil {
+		return err
+	}
+
+	// Reconcile RoleBindings.
+	if organizationRoleTemplate.HasBindTo(corev1alpha1.BindToOwners) {
+		roleBinding := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      organizationRoleTemplate.Name,
+				Namespace: organization.Status.Namespace.Name,
+			},
+			Subjects: organization.Spec.Owners,
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "Role",
+				Name:     role.Name,
+			},
+		}
+		if err := r.reconcileRoleBinding(ctx, roleBinding, organizationRoleTemplate); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *OrganizationRoleTemplateReconciler) reconcileRole(ctx context.Context, role *rbacv1.Role, organizationRoleTemplate *corev1alpha1.OrganizationRoleTemplate) error {
+	desiredRole := role.DeepCopy()
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, role, func() error {
+		if err := controllerutil.SetControllerReference(
+			organizationRoleTemplate, role, r.Scheme); err != nil {
+			return fmt.Errorf("set controller reference for Role: %w", err)
+		}
+		if !reflect.DeepEqual(role.Rules, desiredRole.Rules) {
```

Comment:
why do you even check `!reflect.DeepEqual` and not just setting it up every time?

Reply:
https://github.com/kubermatic/bulward/pull/31#discussion_r450186842

Reply:
See `CreateOrUpdate` source code, it checks for you after the mutate fn finishes.

Reply:
You are right, I misunderstood your previous comment